test/testcafe/spec/IdentifyRecoveryWithCaptcha_spec.js

```diff
@@ -0,0 +1,94 @@
+import { RequestMock, RequestLogger, Selector } from 'testcafe';
+import IdentityPageObject from '../framework/page-objects/IdentityPageObject';
+import { checkConsoleMessages } from '../framework/shared';
+import xhrIdentifyRecoveryWithRecaptcha from '../../../playground/mocks/data/idp/idx/identify-recovery-with-recaptcha-v2';
+import xhrIdentifyRecoveryWithHcaptcha from '../../../playground/mocks/data/idp/idx/identify-recovery-with-hcaptcha';
+import xhrAuthenticatorVerifySelect from '../../../playground/mocks/data/idp/idx/authenticator-verification-select-authenticator';
+
+const identifyRecoveryWithReCaptchaMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrIdentifyRecoveryWithRecaptcha)
+  .onRequestTo('http://localhost:3000/idp/idx/identify')
+  .respond(xhrAuthenticatorVerifySelect);
+
+const identifyRecoveryWithHCaptchaMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrIdentifyRecoveryWithHcaptcha)
+  .onRequestTo('http://localhost:3000/idp/idx/identify')
+  .respond(xhrAuthenticatorVerifySelect);
+
+const identifyRequestLogger = RequestLogger(
+  /idx\/identify/,
+  {
+    logRequestBody: true,
+    stringifyRequestBody: true,
+  }
+);
+
+const reCaptchaRequestLogger = RequestLogger(
+  /\/recaptcha\/api2\/userverify/,
+  {
+    logRequestBody: true,
+    stringifyRequestBody: true,
+  }
+);
+
+fixture('Identify Recovery - reset flow with Captcha');
+
+async function setup(t) {
+  const identityPage = new IdentityPageObject(t);
+  await identityPage.navigateToPage();
+  await checkConsoleMessages({
+    controller: 'forgot-password',
+    formName: 'identify-recovery',
+  });
+
+  return identityPage;
+}
+
+test.requestHooks(identifyRequestLogger, reCaptchaRequestLogger, identifyRecoveryWithReCaptchaMock)('should be able to submit identifier with reCaptcha enabled', async t => {
+  const identityPage = await setup(t);
+  
+  await identityPage.fillIdentifierField('test.identifier');
+  
+  // Wait for the reCaptcha container to appear in the DOM and become visible.
+  await t.expect(Selector('#captcha-container').find('.grecaptcha-badge').exists).ok({timeout: 3000});
+
+  await identityPage.clickNextButton();
+
+  await t.wait(3000);
```

Comment:
We should not need this wait..

Reply:
We do need this because otherwise in some cases the call to the reCaptcha API will not get registered in our request logger fast enough, and the line immediately after this could fail - causing flakiness. In my local tests, I noticed 1s wasn't enough all the time, so I increased it to 3s so it avoids any flakiness.

Reply:
Never a good fan of introducing `wait`..  what screen shows up after you click next button ?

Can we check for some other element after hitting submit and before checking t.expect(reCaptchaRequestLogger.count(() => true)).eql(1);

Reply:
transferred this conversation over to Slack